service/collector/endpoint.go

```diff
@@ -47,23 +61,212 @@ func NewEndpoint(config EndpointConfig) (*Endpoint, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
 
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		MaxIdleConns:    maxIdleConnection,
+	}
+
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   maxTimeoutSec * time.Second,
+	}
+
 	i := &Endpoint{
-		g8sClient: config.G8sClient,
-		k8sClient: config.K8sClient,
-		logger:    config.Logger,
+		g8sClient:     config.G8sClient,
+		httpClient:    client,
+		httpTransport: tr,
+		k8sClient:     config.K8sClient,
+		localIP:       getLocalIP(),
+		logger:        config.Logger,
 	}
 
 	return i, nil
 }
 
-func (i *Endpoint) Collect(ch chan<- prometheus.Metric) error {
+func (e *Endpoint) Collect(ch chan<- prometheus.Metric) error {
 	// TODO
 	// Here we will put implementation of collecting data for each endpoint IP
+	listOpts := metav1.ListOptions{}
+	getOpts := metav1.GetOptions{}
+
+	kvmConfigs, err := e.g8sClient.ProviderV1alpha1().KVMConfigs(crdNamespace).List(listOpts)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, kvmConfig := range kvmConfigs.Items {
```

Comment:
I'd maybe spin a separate goroutine for each item so that following blocking k8s API call occurs concurrently. Check https://godoc.org/golang.org/x/sync/errgroup that we use quite a lot for this kind of cases. E.g. IPAM in aws-operator: https://github.com/giantswarm/aws-operator/blob/master/service/controller/legacy/v27/resource/ipam/create.go#L92-L158

Reply:
done